src/vanilla/utils/atomWithStorage.ts

```diff
@@ -114,6 +129,42 @@ export function createJSONStorage<Value>(
 ): AsyncStorage<Value> | SyncStorage<Value> {
   let lastStr: string | undefined
   let lastValue: Value
+
+  const webStorageSubscribe: Subscribe<Value> = (key, callback) => {
```

Comment:
This is still string based, right?
```suggestion
  const webStorageSubscribe: Subscribe<string> = (key, callback) => {
```

Reply:
before this PR, subscribe method was defined like this:

```ts
subscribe?: (
    key: string,
    callback: (value: Value) => void,
    initialValue: Value,
  ) => Unsubscribe
```
and the callback would accept value of type `Value`. should we change this to accept only strings?

Reply:
hey @dai-shi. I finally got some time to work on this. I would appreciate your opinion (and advice) on typing issues.

Regarding this code suggestion, `webStorageSubscribe` was previously typed as `Subscribe<Value>` and I didn't change the type here. how should we address these typings? Isn't the `Value` type always a subset of `string`?